### PR TITLE
Switched styling to use material design

### PR DESCRIPTION
**Pull request by BenjaminPalko**

### CODE REVIEW

File: components/board.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from "react";
 import Square from "./square";
+import {Box, Grid} from "@mui/material";
 
 interface BoardProps {
```
Comment (piopi) [inline components/board.tsx:6]: I would convert this class to a react component function
```
const Board = () => { return (...)}
```

File: components/board.tsx [hunk 2]
```diff
@@ -48,24 +49,18 @@ export default class Board extends React.Component<BoardProps, BoardState> {
...
+                <Grid container spacing={1}>
+                    {
+                        this.state.squares.map((value, index) => {
+                            return (<Grid item key={index} xs={4}>
+                                {this.renderSquare(index, value)}
+                            </Grid>)
+                        })
+                    }
+                </Grid>
+            </Box>
```
Comment (piopi) [inline components/board.tsx:63]:
```suggestion
            <Box>
                <div>{status}</div>
                <Grid container spacing={1}>
                    {
                        this.state.squares.map((value, index) => (
                            <Grid item key={index} xs={4}>
                               <Square onClick={() => this.handleClick(i)} value={value} />
                            </Grid>)
                        ))
                    }
                </Grid>
            </Box>
```

I would also avoid using index a key it is considered an anti-pattern, https://robinpokorny.com/blog/index-as-a-key-is-an-anti-pattern/

File: components/game.tsx [hunk 3]
```diff
@@ -6,14 +6,6 @@ interface GameProps {
 
 export default function Game(props: GameProps) {
```
Comment (piopi) [inline components/game.tsx:7]: Do you need props for later? if not you can remove them

File: components/square.tsx [hunk 4]
```diff
@@ -1,14 +1,30 @@
+import {Button, Skeleton} from "@mui/material";
+import {Close, RadioButtonUnchecked} from "@mui/icons-material";
+
 interface SquareProps {
     onClick: () => void;
     value: any;
 }
 
 export default function Square(props: SquareProps) {
```
Comment (piopi) [inline components/square.tsx:9]: `export default function Square({onClick, value}: SquareProps)`

Destructuring will avoid using props.value

File: components/square.tsx [hunk 5]
```diff
@@ -1,14 +1,30 @@
...
-            onClick={() => props.onClick()}>
-            {props.value}
-        </button>
+        <Button
+            disabled={props.value}
+            variant={"outlined"}
+            onClick={() => props.onClick()}
+        fullWidth={true}>
+            {getIcon()}
+        </Button>
```
Comment (piopi) [inline components/square.tsx:28]:
```suggestion
        <Button
            disabled={props.value}
            variant={"outlined"}
            onClick={() => props.onClick()}
        fullWidth={true}>
            {!props.value && <Skeleton variant={"text"} width={16} animation={false} />}
            {props.value === 'X' && <Close />}
            {props.value === 'O' && <RadioButtonUnchecked />}
        </Button>
```

This one is just a preference but if you wanna use a function you should combine it with useCallback.
https://reactjs.org/docs/hooks-reference.html#usecallback

File: pages/index.tsx [hunk 6]
```diff
@@ -1,10 +1,22 @@
...
-  )
+    return (
+        <Container maxWidth={"sm"}>
+            <Paper style={{padding: 16}}>
+                <Typography
+                    variant={"h2"}
+                    style={{textAlign: "center"}}
+                    gutterBottom
+                >
+                    Tic-Tac-Toe
```
Comment (piopi) [inline pages/index.tsx:14]: I would store this string a variable

File: pages/index.tsx [hunk 7]
```diff
@@ -1,10 +1,22 @@
...
 const Home: NextPage = () => {
-  return (
-    <Game />
-  )
+    return (
+        <Container maxWidth={"sm"}>
+            <Paper style={{padding: 16}}>
+                <Typography
+                    variant={"h2"}
+                    style={{textAlign: "center"}}
```
Comment (piopi) [inline pages/index.tsx:11]: Small suggestion, I would use the styled utility of Material to add CSS: https://mui.com/system/styled/ , it allows you to access your theme variable and create reusable components with the same style, you can override the style of any component or HTML tag

File: styles/globals.css [hunk 8]
```diff
@@ -1,51 +1,3 @@
 body {
-  font: 14px "Century Gothic", Futura, sans-serif;
-  margin: 20px;
+  background-color: #3d3dcc;
```
Comment (piopi) [inline styles/globals.css:2]: color should be stored in a variable if possible

**Review by piopi — CHANGES_REQUESTED** [review]

File: pages/_app.tsx [hunk 9]
```diff
@@ -1,3 +1,7 @@
+import '@fontsource/roboto/300.css';
+import '@fontsource/roboto/400.css';
+import '@fontsource/roboto/500.css';
+import '@fontsource/roboto/700.css';
```
Comment (piopi) [inline pages/_app.tsx:4]: https://mui.com/material-ui/customization/typography/

File: components/game.tsx (hunk 3 above)
Comment (BenjaminPalko) [reply components/game.tsx:7]: Yeah we are most definitely going to need props on Game

File: pages/index.tsx (hunk 7 above)
Comment (BenjaminPalko) [reply pages/index.tsx]: Ended up just using sx on each component since they are small changes

**Pull request merged**